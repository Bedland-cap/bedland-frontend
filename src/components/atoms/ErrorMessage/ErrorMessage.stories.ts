import type { Meta, StoryObj } from '@storybook/react';
import ErrorMessage from './ErrorMessage';

const meta = {
  title: 'Basic Components/Atoms/ErrorMessage',
  component: ErrorMessage,
  tags: ['autodocs'],
} satisfies Meta<typeof ErrorMessage>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    children: 'error',
  },
};
