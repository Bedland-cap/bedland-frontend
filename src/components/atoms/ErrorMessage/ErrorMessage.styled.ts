import styled from 'styled-components';

const ErrorMsg = styled.span`
  align-items: center;
  color: red;
  display: flex;
  flex-direction: row;
  gap: 5px;
  height: 2.875rem;
`;

export default ErrorMsg;
