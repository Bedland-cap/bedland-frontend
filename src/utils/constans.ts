export const HEADER_HEIGHT = 70;
export const FOOTER_HEIGHT = 52;
