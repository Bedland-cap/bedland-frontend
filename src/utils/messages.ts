const ERROR_MESSAGES = {
  resetPasswordPage: {
    notWithPasswordsRules:
      'Password is incorrect, please try again after reviewing the rules above',

    passwordsDoesNotMatchUp: 'Passwords do not match to each other',
  },
};

export default ERROR_MESSAGES;
