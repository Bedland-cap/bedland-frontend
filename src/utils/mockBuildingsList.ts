import { Building } from 'store/reducers/building/building_slice';

const mockBuildingsList: Building[] = [
  {
    id: '1',
    name: 'Berry',
    address: 'Berry Street 21',
    managerId: '1',
    floors: '5',
  },
  {
    id: '2',
    name: 'Miedziana',
    address: 'Miedziana Street 3',
    managerId: '1',
    floors: '7',
  },
  {
    id: '3',
    name: 'Seaport',
    address: 'Seaport Avenue 51',
    managerId: '2',
    floors: '5',
  },
  {
    id: '4',
    name: 'Desert',
    address: 'Desert Street 2X',
    managerId: '1',
    floors: '6',
  },
  {
    id: '5',
    name: 'Lakewood',
    address: 'Lakewood Avenue 57',
    managerId: '3',
    floors: '10',
  },
];

export default mockBuildingsList;
