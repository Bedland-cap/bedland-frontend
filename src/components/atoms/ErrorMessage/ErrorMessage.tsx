import { FC, PropsWithChildren } from 'react';

import Icon from '../Icon/Icon';
import Typography from '../Typography/Typography';
import * as Styled from './ErrorMessage.styled';

const ErrorMessage: FC<PropsWithChildren> = ({ children }) => (
  <Styled.ErrorRow>
    <div style={{ display: 'flex' }}>
      <Icon name='alert' color='danger' size={20} isActive={false} />
    </div>
    <Typography variant='error' color='danger'>
      {children}
    </Typography>
  </Styled.ErrorRow>
);

export default ErrorMessage;
