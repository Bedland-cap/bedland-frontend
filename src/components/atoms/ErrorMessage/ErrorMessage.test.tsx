import { renderWithProviders } from 'test.utils';
import ErrorMessage from './ErrorMessage';

describe('ErrorMessage', () => {
  it('renders correctly', () => {
    const snapshot = renderWithProviders(<ErrorMessage>Error message</ErrorMessage>, {});
    expect(snapshot).toMatchSnapshot();
  });
});
