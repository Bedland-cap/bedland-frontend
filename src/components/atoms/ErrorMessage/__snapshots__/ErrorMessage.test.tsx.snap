// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage renders correctly 1`] = `
<span
  className="sc-bcXHqe cAICJi"
>
  Error message
</span>
`;
